modules/github-bots/sdk/bot.go

```diff
@@ -90,6 +99,15 @@ func Serve(b Bot) {
 			}
 		}()
 
+		// loop over all event headers and add them to the context so they can be used by the handlers
+		for k, v := range event.Context.GetExtensions() {
+			ctx = context.WithValue(ctx, contextKey(k), v)
+		}
+
+		// add existing event attributes to context so they can be used by the handlers
+		ctx = context.WithValue(ctx, ContextKeyAttributes, event.Extensions())
+		ctx = context.WithValue(ctx, ContextKeyType, event.Type())
```

Comment:
can we scooch this down right before the handler `switch` so we're not needlessly processing events that ultimately get ignored?

Reply:
yeah good shout...